### PR TITLE
[llvm-readelf] Print a blank line for the first hex/string dump

### PR DESCRIPTION
**Pull request by MaskRay**
GNU readelf prints a blank line before the first hex/string dump, which
serves as a separator when there are other dump operations. Port the
behavior.

### CODE REVIEW

File: llvm/test/tools/llvm-readobj/ELF/hex-dump-multi.s [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # REQUIRES: x86-registered-target
 
 # RUN: llvm-mc -filetype=obj -triple x86_64 %s -o %t.o
-# RUN: llvm-readobj -x .a -x .b %t.o | FileCheck %s
+# RUN: llvm-readobj -x .a -x .b %t.o | FileCheck %s --check-prefixes=HEADER,CHECK
 # RUN: llvm-readelf -x .a -x .b %t.o | FileCheck %s
 
-# CHECK:      Hex dump of section '.a':
+# HEADER:     LoadName:
```
Comment (jh7370) [inline llvm/test/tools/llvm-readobj/ELF/hex-dump-multi.s:7]: Is the HEADER check here actually important to the test? It feels to me like you could do the same as in hex-dump.s and just check the blank line.

File: llvm/test/tools/llvm-readobj/ELF/hex-dump.test [hunk 2]
```diff
@@ -46,7 +46,8 @@ FileHeader:
 # RUN: llvm-readelf --hex-dump=.sec %t2.out1 | \
 # RUN:   FileCheck %s --match-full-lines --strict-whitespace --check-prefix=SPACES1
 
-#      SPACES1:Hex dump of section '.sec':
+#      SPACES1:{{^$}}
```
Comment (jh7370) [inline llvm/test/tools/llvm-readobj/ELF/hex-dump.test:49]: Perhaps for consistency, it's worth adding this check to the other cases below?

File: llvm/test/tools/llvm-readobj/ELF/string-dump-multi.s [hunk 3]
```diff
@@ -1,10 +1,12 @@
 # REQUIRES: x86-registered-target
 
 # RUN: llvm-mc -filetype=obj -triple x86_64 %s -o %t.o
-# RUN: llvm-readobj -p .a -p .b %t.o | FileCheck %s
+# RUN: llvm-readobj -p .a -p .b %t.o | FileCheck %s --check-prefixes=HEADER,CHECK
 # RUN: llvm-readelf -p .a -p .b %t.o | FileCheck %s
 
-# CHECK:      String dump of section '.a':
+# HEADER:     LoadName:
```
Comment (jh7370) [inline llvm/test/tools/llvm-readobj/ELF/string-dump-multi.s:7]: Same comment as above.

File: llvm/test/tools/llvm-readobj/ELF/string-dump.test [hunk 4]
```diff
@@ -23,7 +23,9 @@
 # RUN: llvm-readelf -hp1 -p2 %t | cmp %t.readelf.out -
 # RUN: llvm-readelf -hp 1 -p.not_null_terminated %t | cmp %t.readelf.out -
 
-# CHECK:      String dump of section '.strings':
+# HEADER:     LoadName:
```
Comment (jh7370) [inline llvm/test/tools/llvm-readobj/ELF/string-dump.test:26]: Same as above.

**Review by jh7370 — APPROVED** [review]
LGTM, with some minor test comments.

File: llvm/test/tools/llvm-readobj/ELF/hex-dump-multi.s (hunk 1 above)
Comment (MaskRay) [reply llvm/test/tools/llvm-readobj/ELF/hex-dump-multi.s:7]: `llvm-readobj -x .a -x .b` prints two blank lines before the first `Hex dump of section`.
We have to test `LoadName:` to skip the first blank line (not due to hex/string dump).

File: llvm/test/tools/llvm-readobj/ELF/string-dump.test (hunk 4 above)
Comment (MaskRay) [reply llvm/test/tools/llvm-readobj/ELF/string-dump.test:26]: llvm-readobj -x .a -x .b prints two blank lines before the first Hex dump of section.
We have to test `LoadName:` to skip the first blank line (not due to hex/string dump).

**Pull request merged**